### DISCUSSION TITLE
Silence macOS 14 warning

### DISCUSSION THREAD

**Discussion by alberto-lazari** [Homebrew/discussions#4739] [Everyday usage] [answered]
### Output of `brew config`

```
$ /opt/homebrew/bin/brew config
HOMEBREW_VERSION: 4.1.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3cd72905ce3d36a84994cc6c6b9ebbdafff4e922
Last commit: 7 days ago
Core tap JSON: 25 Aug 14:16 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1692336968
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```
$ /opt/homebrew/bin/brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  sphinx

Warning: You are using macOS 14.
We do not provide support for this pre-release version.
It is expected behaviour that some formulae will fail to build in this pre-release version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Is there a way to silence the warning about the pre-release versions of macOS? I could't find any.

I think that homebrew should provide a way to silence it, maybe through a `HOMEBREW_SILENCE_PRE_RELEASE_WARNING` environment variable.

I am aware of the meaning of the message and it makes sense to print it by default and make it difficult to mute it, but it would also make sense to allow users that read it multiple times a day for three months to just get rid of it, at least in `brew upgrade`, without the need to manually grep it out of the way.

**Comment by p-linnane** [comment, marked as answer]
Setting `HOMEBREW_DEVELOPER=1` will silence the message.

**Reply by alberto-lazari** [reply]
Thanks a lot! Where is it documented?

**Reply by p-linnane** [reply]
It's not called out explicitly, but environment variables can be found [here](https://docs.brew.sh/Manpage#environment).

**Reply by alberto-lazari** [reply]
Although it works, I think this is quite a workaround.

I would suggest to move the flag outside of it. I see that the description of the variable says that there are various behaviors changed by this flag:
> HOMEBREW_DEVELOPER
If set, tweak behaviour to be more relevant for Homebrew developers (active or budding) by e.g. turning warnings into errors.

For a normal user like me, not involved in Homebrew development, I have no interest in all the rest (warnings -> errors for instance) so a more specific variable would definitely work better for me (and it would be much more explicit and easy to find).

Maybe it could be something worth opening an issue for?

**Reply by p-linnane** [reply]
It's not really a workaround. If you're on a pre-release OS you need to understand you are going to run into issues whether you are a developer or not. The warning is to keep reminding you of this so that you understand you are responsible for any fixes required on your system. The strong wording of the warning should make it clear how often we maintainers encounter isues like this, so we will not introduce a variable specifically to disable that warning.

**Reply by alberto-lazari** [reply]
Sure, that is clear. Thanks for replying!

**Reply by osalbahr** [reply]
@albertolazari if it makes you feel better, warnings being turned into errors never happened to me and I’ve had developer mode on for quite some time. I’m assuming this shouldn’t happen unless you manually made a change to Homebrew locally. The only practical difference you’ll probably notice is that you’ll be using Homebrew on the master branch rather than the latest release. The master branch is pretty stable in my experience due to all the reviews and CI testing.

**Reply by alberto-lazari** [reply]
Thanks, I'll switch back to normal `HOMEBREW_DEVELOPER=0` soon anyway, when official support for Sonoma will come, so I hope not to encounter problems in this small amount of time.

Thanks to both of you for the support!

**Reply by osalbahr** [reply]
FYI, `HOMEBREW_DEVELOPER=0` also counts as "setting" `HOMEBREW_DEVELOPER`. The `=1` is just for good measures. Here is a way to test that:

```console
$ brew developer
Developer mode is enabled.
$ brew developer off
$ brew developer    
Developer mode is disabled.
$ HOMEBREW_DEVELOPER=1 brew developer
Developer mode is enabled because HOMEBREW_DEVELOPER is set.
$ HOMEBREW_DEVELOPER=0 brew developer
Developer mode is enabled because HOMEBREW_DEVELOPER is set.
```

**Reply by alberto-lazari** [reply]
Ok yes, I meant more 'I will remove the line where I set the variable on my `.zprofile`'. I could imagine that it's one of those cases where it's only tested whether the variable is set or not.
